06_puppy/n0npax/memStore.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"errors"
+)
+
+func NewMemStore() MemStore {
+	return MemStore{}
+}
+
+func (m MemStore) CreatePuppy(puppy *Puppy) int {
+	id := len(m)
+	m[id] = puppy
+	return id
+}
+
+func (m MemStore) ReadPuppy(id int) (*Puppy, error) {
+	puppy, ok := m[id]
```

Comment:
If you flip the logic here, you can save a line of code.

```go
if puppy, ok := m[id]; ok {
    return puppy, nil
}
return nil, errors.New("doesn't exist")
```

Reply:
fixed. Thanks